resources/compute_virtual_machines.go

```diff
@@ -0,0 +1,608 @@
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Identity.PrincipalID"),
+			},
+			{
+				Name:        "identity_tenant_id",
+				Description: "The tenant id associated with the virtual machine This property will only be provided for a system assigned identity",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Identity.TenantID"),
+			},
+			{
+				Name:        "identity_type",
+				Description: "The type of identity used for the virtual machine The type 'SystemAssigned, UserAssigned' includes both an implicitly created identity and a set of user assigned identities The type 'None' will remove any identities from the virtual machine Possible values include: 'ResourceIdentityTypeSystemAssigned', 'ResourceIdentityTypeUserAssigned', 'ResourceIdentityTypeSystemAssignedUserAssigned', 'ResourceIdentityTypeNone'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Identity.Type"),
+			},
+			{
+				Name:        "identity_user_assigned_identities",
+				Description: "The list of user identities associated with the Virtual Machine The user identity dictionary key references will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/MicrosoftManagedIdentity/userAssignedIdentities/{identityName}'",
+				Type:        schema.TypeJSON,
+				Resolver:    schema.PathResolver("Identity.UserAssignedIdentities"),
+			},
+			{
+				Name:        "zones",
+				Description: "The virtual machine zones",
+				Type:        schema.TypeStringArray,
+			},
+			{
+				Name:        "extended_location_name",
+				Description: "The name of the extended location",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ExtendedLocation.Name"),
+			},
+			{
+				Name:        "extended_location_type",
+				Description: "The type of the extended location Possible values include: 'EdgeZone'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ExtendedLocation.Type"),
+			},
+			{
+				Name:        "id",
+				Description: "Resource Id",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ID"),
+			},
+			{
+				Name:        "name",
+				Description: "Resource name",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "type",
+				Description: "Resource type",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "location",
+				Description: "Resource location",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "tags",
+				Description: "Resource tags",
+				Type:        schema.TypeJSON,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:         "azure_compute_virtual_machine_win_config_rm_listeners",
+				Description:  "WinRMListener describes Protocol and thumbprint of Windows Remote Management listener",
+				Resolver:     fetchComputeVirtualMachineWinConfigRmListeners,
+				AlwaysDelete: true,
```

Comment:
upgrade to RC5 and you can avoid this, also running always delete without a filter won't work